api/src/main/java/org/openmrs/module/htmlformentryui/element/ObsFromFragmentElement.java

```diff
@@ -0,0 +1,200 @@
+package org.openmrs.module.htmlformentryui.element;
+
+import java.net.URISyntaxException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+import org.openmrs.Concept;
+import org.openmrs.ConceptAnswer;
+import org.openmrs.ConceptDatatype;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.module.htmlformentry.FormEntryContext;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+import org.openmrs.module.htmlformentry.FormSubmissionError;
+import org.openmrs.module.htmlformentry.HtmlFormEntryUtil;
+import org.openmrs.module.htmlformentry.action.FormSubmissionControllerAction;
+import org.openmrs.module.htmlformentry.element.HtmlGeneratorElement;
+import org.openmrs.module.htmlformentryui.tag.UiIncludeTagHandler;
+import org.openmrs.ui.framework.FragmentException;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.page.PageAction;
+
+public class ObsFromFragmentElement implements HtmlGeneratorElement, FormSubmissionControllerAction {
+
+	private UiUtils uiUtils;
+	private Map<String, Object> fragmentParams;
+	private Concept concept;
+	private String viewProvider;
+	private String fragment;
+	private FragmentType fragmentType;
+	
+	public ObsFromFragmentElement(Map<String, String> parameters, UiUtils uiUtils) {
+		this.uiUtils = uiUtils;
+		concept = HtmlFormEntryUtil.getConcept(parameters.get("conceptId"));
+		if (concept == null) {
+			throw new IllegalArgumentException("Cannot find concept with id " + parameters.get("conceptId"));
+		}
+		viewProvider = parameters.get("provider");
+		fragment = parameters.get("fragment");
+		StringBuilder fragmentUrlBuilder = new StringBuilder();
+		fragmentUrlBuilder.append(fragment);
+		fragmentUrlBuilder.append("?");
+		fragmentUrlBuilder.append(parameters.get("fragmentParams"));
+		try {
+			fragmentParams = UiIncludeTagHandler.paramsToMap(fragmentUrlBuilder.toString());
+			// first checkup the type from the params
+			if (fragmentParams.get("type") != null) {
+				String type = (String) fragmentParams.get("type");
```

Comment:
Is this a design assumption of some sort, that a `"type"` will _have_ to be provided as part f the fragment params?